### PR TITLE
Update worker exception

### PR DESCRIPTION
**Pull request by jja725**
### What changes are proposed in this pull request?

Replace checked exception with run time exception in worker when appropriate.

### Why are the changes needed?

code quality

### Does this PR introduce any user facing changes?

na

### CODE REVIEW

File: core/common/src/main/java/alluxio/util/io/FileUtils.java [hunk 1]
```diff
@@ -71,16 +74,20 @@ public static void changeLocalFileGroup(String path, String group) throws IOExce
    * @param filePath that will change permission
    * @param perms the permission, e.g. "rwxr--r--"
    */
-  public static void changeLocalFilePermission(String filePath, String perms) throws IOException {
-    Files.setPosixFilePermissions(Paths.get(filePath), PosixFilePermissions.fromString(perms));
+  public static void changeLocalFilePermission(String filePath, String perms) {
+    try {
+      Files.setPosixFilePermissions(Paths.get(filePath), PosixFilePermissions.fromString(perms));
+    } catch (IOException e) {
```
Comment (rongrong) [inline core/common/src/main/java/alluxio/util/io/FileUtils.java:80]: You can catch specific exceptions as defined in the function API and handle them here. the implementation of `AlluxioRuntimeException.from` should be simple.

File: core/common/src/main/java/alluxio/util/io/FileUtils.java [hunk 2]
```diff
@@ -302,17 +308,17 @@ public FileVisitResult postVisitDirectory(Path dir, IOException e) throws IOExce
...
+  public static boolean createStorageDirPath(String path, String workerDataFolderPermissions) {
     if (Files.exists(Paths.get(path))) {
       return false;
     }
     Path storagePath;
     try {
       storagePath = Files.createDirectories(Paths.get(path));
-    } catch (UnsupportedOperationException | SecurityException | IOException e) {
-      throw new IOException("Failed to create folder " + path, e);
+    } catch (IOException e) {
```
Comment (rongrong) [inline core/common/src/main/java/alluxio/util/io/FileUtils.java:318]: Same here, catch & handle specific exceptions.

File: core/common/src/main/java/alluxio/util/io/FileUtils.java [hunk 3]
```diff
@@ -324,13 +330,17 @@ public static boolean createStorageDirPath(String path, String workerDataFolderP
...
    */
-  public static void createFile(String filePath) throws IOException {
+  public static void createFile(String filePath) {
     Path storagePath = Paths.get(filePath);
     Path parent = storagePath.getParent();
-    if (parent != null) {
-      Files.createDirectories(parent);
+    try {
+      if (parent != null) {
+        Files.createDirectories(parent);
```
Comment (rongrong) [inline core/common/src/main/java/alluxio/util/io/FileUtils.java:338]: here as well

File: core/common/src/main/java/alluxio/worker/block/io/LocalFileBlockWriter.java [hunk 4]
```diff
@@ -61,11 +61,19 @@ public LocalFileBlockWriter(String path) {
...
+
+    int inputBufLength = inputBuf.limit() - inputBuf.position();
+    MappedByteBuffer outputBuf;
     try {
-      bytesWritten = write(mLocalFileChannel.size(), inputBuf);
+      outputBuf = mLocalFileChannel.map(FileChannel.MapMode.READ_WRITE, mLocalFileChannel.size(),
+          inputBufLength);
     } catch (IOException e) {
-      throw AlluxioRuntimeException.from(e);
+      throw new InternalRuntimeException(e);
```
Comment (jja725) [inline core/common/src/main/java/alluxio/worker/block/io/LocalFileBlockWriter.java:71]: There's other runtime exception (illegalState and illegalArgument)which already covered by from method.

File: core/common/src/main/java/alluxio/exception/runtime/AlluxioRuntimeException.java [hunk 5]
```diff
@@ -190,9 +163,24 @@ public static AlluxioRuntimeException from(AlluxioStatusException t) {
...
+    }
+    if (ioe instanceof java.nio.file.DirectoryNotEmptyException) {
+      return new FailedPreconditionRuntimeException(ioe);
+    }
+    if (ioe instanceof java.nio.file.AtomicMoveNotSupportedException) {
+      return new UnimplementedRuntimeException(ioe, ErrorType.External);
+    }
+    if (ioe instanceof java.nio.file.AccessDeniedException) {
+      return new PermissionDeniedRuntimeException(ioe);
+    }
```
Comment (rongrong) [inline core/common/src/main/java/alluxio/exception/runtime/AlluxioRuntimeException.java:183]: Is these still necessary here?

File: core/common/src/main/java/alluxio/exception/runtime/BlockDoesNotExistRuntimeException.java [hunk 6]
```diff
@@ -9,29 +9,22 @@
...
-  private static final Status STATUS = Status.NOT_FOUND;
-  private static final ErrorType ERROR_TYPE = ErrorType.User;
+public class BlockDoesNotExistRuntimeException extends NotFoundRuntimeException {
 
   /**
    * Constructs a new exception with the specified detail message.
    *
    * @param message the detail message
    */
   public BlockDoesNotExistRuntimeException(String message) {
```
Comment (rongrong) [inline core/common/src/main/java/alluxio/exception/runtime/BlockDoesNotExistRuntimeException.java:26]: This API doesn't seem to be necessary. If a block does not exist exception takes an arbitrary message, caller might as well use not found directly.

File: core/common/src/main/java/alluxio/exception/runtime/UnavailableRuntimeException.java [hunk 7]
```diff
@@ -9,9 +9,9 @@
  * See the NOTICE file distributed with this work for information regarding copyright ownership.
  */
 
-package alluxio.exception.status;
+package alluxio.exception.runtime;
 
-import alluxio.exception.AlluxioRuntimeException;
+import alluxio.exception.status.FailedPreconditionException;
```
Comment (rongrong) [inline core/common/src/main/java/alluxio/exception/runtime/UnavailableRuntimeException.java:14]: This is referred to in the comment and comment seems to be not relevant?

File: core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java [hunk 8]
```diff
@@ -277,46 +279,38 @@ public void abortBlock(long sessionId, long blockId) throws IOException {
...
       }
-
       InetSocketAddress address =
           InetSocketAddress.createUnresolved(mAddress.getHost(), mAddress.getRpcPort());
-      throw new WorkerOutOfSpaceException(ExceptionMessage.CANNOT_REQUEST_SPACE
-          .getMessageWithUrl(RuntimeConstants.ALLUXIO_DEBUG_DOCS_URL, address, blockId), e);
+      throw new ResourceExhaustedRuntimeException(
+          ExceptionMessage.CANNOT_REQUEST_SPACE.getMessageWithUrl(
+              RuntimeConstants.ALLUXIO_DEBUG_DOCS_URL, address, blockId), e, false);
     }
```
Comment (rongrong) [inline core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java:314]: This logic feels very much unnecessary. We can probably make sure wherever throws this include worker ID or worker address in the error message?

File: core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java [hunk 9]
```diff
@@ -543,12 +529,15 @@ private BlockStoreLocation commitBlockInternal(long sessionId, long blockId, boo
     BlockStoreLocation loc = tempBlockMeta.getBlockLocation();
 
     // Heavy IO is guarded by block lock but not metadata lock. This may throw IOException.
-    FileUtils.move(srcPath, dstPath);
+    try {
+      FileUtils.move(srcPath, dstPath);
```
Comment (rongrong) [inline core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java:533]: This should probably also be changed to handle specific exception so here you don't need to catch IO exception.

**Review by rongrong — APPROVED** [review]

File: core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java (hunk 9 above)
Comment (jja725) [reply core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java:533]: yes, I have a comment that I would have a follow-up PR once I figure out the Move API.

File: core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java (hunk 8 above)
Comment (jja725) [reply core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java:314]: I agree, probably need to include this information in some context class and use it at the bottom level.

**Comment by rongrong** [comment]
alluxio-bot, merge this please

**Pull request merged**